### DISCUSSION TITLE
HELP Workflow mask for inpaint

### DISCUSSION THREAD

**Discussion by kalle07** [Acly/krita-ai-diffusion#1347] [Q&A] [answered]
iam prof in Adobe photoshop but here i dont get it even with the help of
https://docs.krita.org/en/user_manual/selections.html#display-modes

and it should be at most two clicks an draw a mask

![378756235-44a71e32-9364-4b4b-a4f2-eca05e2c28fe](https://github.com/user-attachments/assets/22af443e-69c6-494e-8c36-93c13fcf9456)

**Comment by markus1921** [comment]
hi, how did you do it? if you add a transparency mask there should be no problem

**Comment by kalle07** [comment]
![grafik](https://github.com/user-attachments/assets/1ff80f48-1d20-4bd1-8fe0-1b3316c5afb0)


![grafik](https://github.com/user-attachments/assets/cd8b7940-ae7e-4fae-bdee-0d5346c614b1)


but it does not recognize that ... for me only works selection but thats sometimes not usefuly

**Comment by kalle07** [comment]
HELP   ;)

**Comment by kalle07** [comment]
Help ?

**Comment by Drakosha405** [comment]
ctrl + lmb on  transparency mask => ctrl+shift+I(invert mask select).

Select menu => show global mask.  select with tool  then select new layer and with brush draw shape.

create new layer paint with color shape => ctrl+lmb on this layer.

this addon work whit select/ if u want use  paint mask (new layer) and use it without select u need create your own workflow/

**Comment by kalle07** [comment]
so thats what i mean it is TOTALY complicated !!!
every inpaint option in any know program like forge, a1111, swarmui, invoke is one click!

so only options is selection : D

i start to post this as more easy feature

**Comment by Drakosha405** [comment]
one click!
![select](https://github.com/user-attachments/assets/062ed1ae-5454-4eab-94c2-fcaa1f58844b)
all the previous ones describe how to make the addon work from what you have shown. and in other programs you do the same thing just in a different wrapper

**Comment by kalle07** [comment]
no i want a mask , you can draw it quickly add a part or mask more than one part of the image ... no selection, sry  ;)

you ever use any inpaint function on another software i mentioned ?


the function and the result are good but hard to get ...

BTW krita is one of the outsiders in function ... very different from ANY other software i know since 20 years
eg. if i open 20 images i choose a tool, and if i take the next image i must choose the tool again ... no other software doo that  ^^

**Reply by kalle07** [reply]
Help ?

**Reply by ETCore7** [reply]
Krita is nice for a free tool, but it can also be a counter intuitive.

Tip: if you have customized dockers, brushes, swatches, libraries and so on make sure to back-up everything. Also try to restore everything at least once on another machine so you can be sure you actually do have a working back-up. Krita overwrites settings every session. If something is wrong it can reset everything to default

**Comment by kalle07** [comment]
I SEE  so no one can give me an explanation  ... i ask me why ?
and you must admit iam not the only stupid   : D

**Comment by Drakosha405** [comment]
you just don't want to understand how it works. the masks that you draw in other programs simply create temporary black and white image files and then process them.  In Krita, masks are used to mix layers. and the selection to which you can add and remove parts creates the conditions for creating a new black and white image for the work of comfyui + gives the exact position for subsequent mixing with the original. this is if explain it in a simple way. and it is not clear your obsession with drawing a mask in general if the selection is much faster and does not create many unnecessary layers (in 95% of cases, a selection with 100% strength is always used. For the remaining 5%, you can also make a couple of extra clicks) https://github.com/Acly/krita-ai-diffusion/wiki/Inpainting (masks are highlighted in color for understanding)
https://docs.krita.org/en/reference_manual/layers_and_masks/selection_masks.html#global-selection
https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.16.0

think of it as windows and linux or  unity vs unreal engine / blender vs 3d max . where you move from one to the other and you need to get used to the new and not look for a way as it was in the old system.
If you can't do it, then you just don't need it.

**Reply by kalle07** [reply]
hmmm, maybe we talk past each other a bit ... 

okay its allways hard to make a selection arround an object and add a part or choose more than one part you must press "shift" and "ctrl"
i want a selection brush !

do you now understand my idea ?

**Comment by Shrapnel24** [comment, marked as answer]
@kalle07

I understand what you are saying about wanting to use a brush to add or subtract from a selection as an alternative to using the selection tool.  Depending on the complexity or precision of the selection you are trying to make it can be a useful alternative. After reading this thread I also became curious if there was an easy method in Krita that also works for how the AI plug-in interprets the selection and think I found an answer that will work for you. It is really pretty simple.

Krita has both Layer Selection Masks and a Global Selection Mask, but the AI plug-in will only recognize the Global Selection as a region in which to do a Fill operation. The nice part is that you can, in fact, use a brush tool to make the selection using the Global Selection Mask.

If you are not already aware, you want to go to Select > Show Global Selection Mask in the Krita menu bar (I personally went into settings and assigned this to my Space Bar for easy toggling). Once this red mask is showing, you will see that there is now a separate (temporary) layer called Selection Mask.  While this layer is selected, you can now use any brush to paint using Black (subtract) or White (add) to define your selection.  You will notice that the Generate button has now been changed to Fill as expected. Once you toggle Show Global Selection Mask back off, the area you marked will have the usual marching ants defining your selection.  You can toggle back and forth this way as much as you like until you deselect.

If you want to save the selection for later use, unfortunately Krita only allows you to save Selection Masks for individual layers and these are not registered as a normal selection by the AI plug-in and there is not an easy built-in way to convert a layer selection mask to a global selection mask.  The most logical way I found is while your selection is made, just make a new layer and fill the selection with the foreground color.  Then when you want to use it again, just select the layer, right-click, and choose Select Opaque (Replace) at the bottom and you will have a global selection that you can toggle the mask for.


Here is the post from the official Krita website where they first describe how this feature works. If you have any other questions about this, you may find the answers here:

https://krita.org/en/posts/2015/how-to-select-areas-with-the-paintbrush-in-krita-2-9/

**Reply by kalle07** [reply]
thats it ... THX!
here it is not well eyplained  ...
https://docs.krita.org/en/user_manual/selections.html#display-modes
